src/libs/actions/Policy.js

```diff
@@ -468,6 +468,75 @@ function update(policyID, values, shouldGrowl = false) {
         });
 }
 
+/**
+ * @param {String} policyID The id of the workspace / policy
+ * @param {String} name The new workspace name
+ * @param {String} currency The new workspace currency
+ */
+function updateWorkspaceGeneralSettings(policyID, name, currency) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    generalSettings: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+
+                // Clear errorFields in case the user didn't dismiss the error
+                errorFields: {
+                    generalSettings: null,
+                },
+                name,
+                outputCurrency: currency,
+            },
+        },
+    ];
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    generalSettings: null,
+                },
+                errorFields: {
+                    generalSettings: null,
+                },
+            },
+        },
+    ];
```

Comment:
Reading over the Pattern B doc, I dont think we need to add the success data, should this not be overwritten once we get online and pusher sends the updated information? https://docs.google.com/document/d/1Ija9CHiNBH1HVnnITA-XuohF31BoiseY18a8_BxFtjc/edit#heading=h.66vmffz9pnn

Reply:
We definitely need to set `pendingFields.generalSettings` to `null` since that is the `pendingAction` prop here
https://github.com/Expensify/App/blob/d4f4187ad8e413a378694a53c12ba71cd5dd0a7e/src/pages/workspace/WorkspaceSettingsPage.js#L150

I suppose I could remove the `errorFields` reset since it's done in the optimistic data.

Reply:
Cool, lets do that! Thanks!

Reply:
Done.